rest-api-spec/src/main/resources/rest-api-spec/api/msearch.json

```diff
@@ -33,6 +33,10 @@
           "type" : "number",
           "description" : "A threshold that enforces a pre-filter roundtrip to prefilter search shards based on query rewriting if the number of shards the search request expands to exceeds the threshold. This filter roundtrip can limit the number of shards significantly if for instance a shard can not match any documents based on it's rewrite method ie. if date filters are mandatory to match but the shard bounds and the query are disjoint.",
           "default" : 128
+        },
+        "group_shard_failures" : {
```

Comment:
with msearch and msearch_template, I had to go with exposing the param only globally. Given how it's read (through `ToXContent.Params` when outputting `SearchResponse` output, it becomes very complicated to support the option per item, and have each response item with its own value read from its own request item.

I also wondered if, in general, it is a good idea to expose this new parameter rather than remove support for it, given that it is not documented, and we did not realize for a long time that it was not exposed, and we did not get any report about this either.